scrapy/settings/__init__.py

```diff
@@ -49,14 +49,13 @@ def __init__(self, value, priority):
 
     def set(self, value, priority):
         """Sets value if priority is higher or equal than current priority."""
-        if isinstance(self.value, BaseSettings):
-            # Ignore self.priority if self.value has per-key priorities
-            self.value.update(value, priority)
-            self.priority = max(self.value.maxpriority(), priority)
-        else:
-            if priority >= self.priority:
+        if priority >= self.priority:
+            if isinstance(self.value, BaseSettings):
+                self.value.clear()
+                self.value.update(value, priority=priority)
```

Comment:
It could be a little odd having a reference for `value` (gotten from `.get()` for example) and having it changed when setting something new, as `set()` was supposed to override the value, not change it. Maybe we could create a new BaseSettings instance?